src/utils/SortingNetworkGenerator.java

```diff
@@ -1,129 +1,266 @@
 package utils;
 
 import java.awt.Desktop;
-import java.io.BufferedReader;
 import java.io.BufferedWriter;
 import java.io.File;
+import java.io.FileOutputStream;
 import java.io.IOException;
-import java.io.InputStreamReader;
 import java.io.OutputStreamWriter;
-import java.lang.ProcessBuilder.Redirect;
-import java.util.stream.Collectors;
-import java.text.DecimalFormat;
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
 
 import javax.swing.JOptionPane;
+import javax.swing.ProgressMonitor;
 
-import frames.UtilFrame;
+import main.ArrayVisualizer;
 import panes.JErrorPane;
-import resources.sorting_network_master.SortingNetworkFetcher;
 
 public class SortingNetworkGenerator {
-    static DecimalFormat formatter = new DecimalFormat();
+    private static final class Comparator {
+        final int i1, i2;
 
-    static boolean hasPython = false;
-    static String pythonCommand = null;
+        Comparator(int i1, int i2) {
+            this.i1 = i1;
+            this.i2 = i2;
+        }
 
-    final static int LIMIT = 20000;
+        @Override
+        public String toString() {
+            return i1 + ":" + i2;
+        }
 
-    static boolean verifyPythonVersion(String minVersion, String command) {
-        try {
-            ProcessBuilder builder = new ProcessBuilder(command, "-c",
-                String.format("import sys; print (sys.version_info >= (%s))", minVersion));
-            Process p = builder.start();
-            if (p.waitFor() != 0) {
-                return false;
-            }
-            BufferedReader r = new BufferedReader(new InputStreamReader(p.getInputStream()));
-            if (r.readLine().equals("True")) {
-                pythonCommand = command;
-                return true;
-            }
-            return false;
+        @Override
+        public int hashCode() {
+            return (i1 << 16) + i2;
         }
-        catch (IOException e) {
-            return false;
+
+        boolean overlaps(Comparator other) {
+            return (this.i1 < other.i1 && other.i1 < this.i2) ||
+                   (this.i1 < other.i2 && other.i2 < this.i2) ||
+                   (other.i1 < this.i1 && this.i1 < other.i2) ||
+                   (other.i1 < this.i2 && this.i2 < other.i2);
```

Comment:
I just copied the code. I'll look at improving it.

Reply:
a small sidenote, these two differ when the two intervals are exactly the same

Reply:
That's probably not a good thing

Reply:
it doesn't matter, since this code is only run for comparators that don't have the same input (as they're part of the same groups)... and the current implementation returns `false` if the two intervals are exactly the same

Reply:
I ran a test, and the changed code produces different results than the original.

Reply:
Oh wait nvm